tools/test-generator/src/main/java/com/consol/citrus/generate/provider/http/ReceiveHttpRequestCodeProvider.java

```diff
@@ -56,7 +55,7 @@ public CodeBlock getCode(String endpoint, HttpMessage message) {
                     code.add(".header($S, $S)\n", entry.getKey(), Optional.ofNullable(entry.getValue()).map(Object::toString).orElse(""));
                 });
 
-        if (!CollectionUtils.isEmpty(message.getQueryParams())) {
+        if (!message.getQueryParams().isEmpty()) {
```

Comment:
This could cause a null pointer, if `message.getQueryParams()` returns `null`. If you change the signature of `message.getQueryParams()` back to a `Map<K, V>` implementation, `CollectionUtils.IsEmpty()` will be usable again, which includes a null check.